.github/workflows/nightly-t5x-build.yaml

```
@@ -22,40 +22,86 @@ permissions:
 jobs:
 
   metadata:
-    if: (github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'success') || github.event_name == 'workflow_dispatch'
     runs-on: ubuntu-22.04
     outputs:
+      PUBLISH: ${{ steps.if-publish.outputs.PUBLISH }}
       BUILD_DATE: ${{ steps.date.outputs.BUILD_DATE }}
     steps:
+      - name: Check if the triggering workflow failed
+        id: if-upstream-failed
+        shell: bash -x -e {0}
+        run: |
+          echo "UPSTREAM_FAILED=${{ github.event.workflow_run.conclusion != 'success' }}" >> $GITHUB_OUTPUT
+
+      - name: Cancel workflow if upstream workflow did not success
+        if: ${{ steps.if-upstream-failed.outputs.UPSTREAM_FAILED == 'true' }}
+        uses: styfle/cancel-workflow-action@0.12.0
+
+      - name: Determine if the resulting container should be 'published'
+        id: if-publish
+        shell: bash -x -e {0}
+        run:
+          # A container should be published if:
+          # 1) the workflow is triggered by workflow_dispatch and the PUBLISH input is true, or
+          # 2) the workflow is triggered by workflow_run (i.e., a nightly build)
+          echo "PUBLISH=${{ github.event_name == 'workflow_run' || (github.event_name == 'workflow_dispatch' && inputs.PUBLISH) }}" >> $GITHUB_OUTPUT
+
       - name: Set build date
         id: date
         shell: bash -x -e {0}
         run: |
           BUILD_DATE=$(TZ='US/Los_Angeles' date '+%Y-%m-%d')
           echo "BUILD_DATE=${BUILD_DATE}" >> $GITHUB_OUTPUT
 
-  build:
-    if: (github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'success') || github.event_name == 'workflow_dispatch'
+  amd64:
     needs: metadata
     uses: ./.github/workflows/_build_t5x.yaml
     with:
+      ARCHITECTURE: amd64
       BUILD_DATE: ${{ needs.metadata.outputs.BUILD_DATE }}
     secrets: inherit
 
-  publish:
-    if: (github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'success') || (github.event_name == 'workflow_dispatch' && inputs.PUBLISH)
-    needs: [metadata, build]
+  arm64:
+    needs: metadata
+    runs-on: ubuntu-22.04
+    outputs:
+      DOCKER_TAG_FINAL: ''
```

Should this be:
```suggestion
      DOCKER_TAG_MEALKIT: ''
      DOCKER_TAG_FINAL: ''
```
?

---

This line was updated, but the change was FINAL -> MEALKIT. Should this not have both outputs?